File: 78-subsets/78-subsets.py
class Solution:
    def func(self,idx,a,arr,res):
        if idx == len(a):
            res.append(list(arr))
            return
        arr.append(a[idx])
        self.func(idx+1,a,arr,res)
        arr.pop()
        self.func(idx+1,a,arr,res)
    
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        self.func(0,nums,[],res)
        return res

File: 78-subsets/78-subsets.cpp
class Solution {
public:
    void solve(vector<int> nums, int n, vector<vector<int>> &ans,int idx, vector<int> &res){
        if(idx==n) {ans.push_back(res); return;} 
        else{
            res.push_back(nums[idx]);
            solve(nums, n, ans, idx+1, res);
            res.pop_back();
            solve(nums, n, ans, idx+1, res);
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> res;
        solve(nums,nums.size(),ans,0,res);
        return ans;
    }
};